paras-marketplace-contract/src/nft_callbacks.rs

```diff
@@ -147,23 +147,20 @@ impl NonFungibleTokenApprovalsReceiver for Contract {
                 price.unwrap().0,
             );
         } else if market_type == "add_trade" {
+            // return;
+            let contract_and_token_id = format!("{}{}{}", nft_contract_id, DELIMETER, token_id);
+            if let Some(mut market_data) = self.market.get(&contract_and_token_id) {
+                market_data.approval_id = approval_id;
+                self.market.insert(&contract_and_token_id, &market_data);
+            }
             let storage_amount = self.storage_minimum_balance().0;
             let owner_paid_storage = self.storage_deposits.get(&signer_id).unwrap_or(0);
             let signer_storage_required =
                 (self.get_supply_by_owner_id(signer_id).0 + 1) as u128 * storage_amount;
 
-            assert!(
-                owner_paid_storage >= signer_storage_required,
-                "Insufficient storage paid: {}, for {} sales at {} rate of per sale",
-                owner_paid_storage,
-                signer_storage_required / storage_amount,
-                storage_amount,
-            );
-
-            let contract_and_token_id = format!("{}{}{}", nft_contract_id, DELIMETER, token_id);
-            if let Some(mut market_data) = self.market.get(&contract_and_token_id) {
-                market_data.approval_id = approval_id;
-                self.market.insert(&contract_and_token_id, &market_data);
+            if owner_paid_storage >= signer_storage_required {
+                env::log_str("Insufficient storage");
```

Comment:
Pakai logging-nya yang "Insufficient storage paid: {}, for {} sales at {} rate of per sale" 

supaya lebih verbose